mne/time_frequency/tests/test_tfr.py

```diff
@@ -680,6 +680,23 @@ def test_getitem_epochsTFR():
     assert_array_equal(power['Trial == "face"'].data,
                        power.data[power.metadata['Trial'] == 'face'])
 
+    # Check that the wrong Key generates a Key Error for Metadata search
     with pytest.raises(KeyError):
         power['Trialz == "place"']
+
+    # Test length function
+    assert len(power) == n_events
+    assert len(power[3:6]) == 3
+
+    ind = 0
+    # Test iteration function
+    for power_ep in power:
+        assert_array_equal(power_ep, power.data[ind])
+        ind += 1
+        if ind == 5:
+            break
```

Comment:
`for ii, power_ep in enumerate(power):`

or

`for ii, power in zip(range(5), power):` (to save the break line)